domain/unitstate/service/service.go

```diff
@@ -42,6 +43,12 @@ type State interface {
 	// SetUnitStateRelation replaces the agent relation
 	// state for the unit with the input UUID.
 	SetUnitStateRelation(domain.AtomicContext, string, map[int]string) error
+
+	// GetUnitState returns the full agent state of the unit.
+	// If no unit with the uuid exists, a [errors.UnitNotFound] error is
+	// returned.
+	// If the units state is empty [errors.EmptyUnitState] error is returned.
```

Comment:
nit: implementation detail, not a description needed in the interface.

Reply:
I would argue that in general if specific error types are returned from the function then they make up part its public interface and should be documented here. Though in this case since we have the GetUnitUUIDForName which returns this error it is not that useful that it is returned here. I have removed it from the description.